RxJavaTemplate/app/src/androidTest/java/co/nimblehq/rxjava/ui/screens/home/HomeFragmentTest.kt

```diff
@@ -0,0 +1,159 @@
+
+        unregisterIdlingResource()
+    }
+
+    @Test
+    fun when_clicking_on_refresh_button_should_refresh_data() {
+        registerIdlingResource()
+
+        unregisterIdlingResource()
+
+        onView(withId(R.id.btHomeRefresh)).perform(click())
+
+        onView(withId(R.id.pbLoading)).check(matches(isDisplayed()))
+        onView(withId(R.id.btHomeRefresh)).check(matches(not(isEnabled())))
+
+        registerIdlingResource()
+
+        onView(withId(R.id.pbLoading)).check(matches(not(isDisplayed())))
+        onView(withId(R.id.btHomeRefresh)).check(matches(isEnabled()))
+
+        unregisterIdlingResource()
+    }
+
+    @Test
+    fun when_clicking_on_item_should_navigate_to_Second_screen() {
+        registerIdlingResource()
+
+        onView(withId(R.id.rvHomeData))
+            .perform(actionOnItemAtPosition<DataAdapter.ViewHolder>(0, click()))
+
+        onView(withId(R.id.tvSecondTitle)).check(matches(withText("title1")))
+        onView(withId(R.id.tvSecondAuthor)).check(matches(withText("author1")))
+
+        pressBack()
+
+        onView(withId(R.id.rvHomeData))
+            .perform(actionOnItemAtPosition<DataAdapter.ViewHolder>(1, scrollTo()))
+            .perform(actionOnItemAtPosition<DataAdapter.ViewHolder>(1, click()))
+
+        onView(withId(R.id.tvSecondTitle)).check(matches(withText("title2")))
+        onView(withId(R.id.tvSecondAuthor)).check(matches(withText("author2")))
+
+        pressBack()
+
+        onView(withId(R.id.rvHomeData))
+            .perform(actionOnItemAtPosition<DataAdapter.ViewHolder>(2, scrollTo()))
+            .perform(actionOnItemAtPosition<DataAdapter.ViewHolder>(2, click()))
+
+        onView(withId(R.id.tvSecondTitle)).check(matches(withText("title3")))
+        onView(withId(R.id.tvSecondAuthor)).check(matches(withText("author3")))
+
+        pressBack()
+
+        onView(withId(R.id.rvHomeData)).check(matches(isDisplayed()))
+
+        unregisterIdlingResource()
+    }
+
+    @Test
+    fun when_clicking_on_open_post_button_should_navigate_to_WebView_screen() {
+        registerIdlingResource()
+
+        onView(withId(R.id.rvHomeData))
+            .perform(actionOnItemAtPosition<DataAdapter.ViewHolder>(0, click()))
+        onView(withId(R.id.btOpenPost)).perform(click())
+
+        onWebView()
+            .withNoTimeout()
+            .check(webMatches(getCurrentUrl(), containsString("google.com")))
+
+        pressBack()
+
+        onView(withId(R.id.ivSecondThumbnail)).check(matches(isDisplayed()))
+
+        pressBack()
+
+        onView(withId(R.id.rvHomeData)).check(matches(isDisplayed()))
+    }
+
+    private fun registerIdlingResource() {
+        IdlingRegistry.getInstance().register(IdlingResource.countingIdlingResource)
+    }
+
+    private fun unregisterIdlingResource() {
+        IdlingRegistry.getInstance().unregister(IdlingResource.countingIdlingResource)
+    }
```

Comment:
maybe we can create helper function then putting it in helper class. How do you think? 🙏 
```suggestion
    private fun setIdlingResource(onIdling: () -> Unit) {
        IdlingRegistry.getInstance().register(IdlingResource.countingIdlingResource)
        onIdling.invoke()
        IdlingRegistry.getInstance().unregister(IdlingResource.countingIdlingResource)
    }
```
then calling:
```dart
setIdlingResource {
  onView(..)...
}
```

Reply:
Agree krub, now it a lot cleaner 🙏🏼